infra/Pos/Subscription/Common.hs

```diff
@@ -86,9 +86,12 @@ subscriptionListener nodeType = listenerConv @Void $ \__ourVerInfo nodeId conv -
     mbMsg <- recvLimited conv
     whenJust mbMsg $ \MsgSubscribe -> do
       let peers = simplePeers [(nodeType, nodeId)]
-      bracket_ (updatePeersBucket BucketSubscriptionListener (<> peers))
-               (updatePeersBucket BucketSubscriptionListener (removePeer nodeId))
-               (void $ recvLimited conv)
+      bracket
+        (updatePeersBucket BucketSubscriptionListener (<> peers))
+        (\added -> when added $
+          void $ updatePeersBucket BucketSubscriptionListener (removePeer nodeId))
+        (\added -> when added $
+          void $ recvLimited conv) -- if not added, close the conversation
```

Comment:
Perhaps we should put some logging in here, particularly for the case where a subscriber can't be added.